src/spi/blocking.rs

```diff
@@ -91,18 +70,46 @@ pub enum Operation<'a, Word: 'static + Copy = u8> {
     /// Write data out while reading data into the provided buffer
     Transfer(&'a mut [Word], &'a [Word]),
     /// Write data out while reading data into the provided buffer
-    TransferInplace(&'a mut [Word]),
+    TransferInPlace(&'a mut [Word]),
 }
 
-/// Transactional trait allows multiple actions to be executed
-/// as part of a single SPI transaction
-pub trait Transactional<Word: 'static + Copy = u8>: ErrorType {
-    /// Execute the provided transactions
-    fn exec<'a>(&mut self, operations: &mut [Operation<'a, Word>]) -> Result<(), Self::Error>;
+/// Blocking read-write SPI
+pub trait ReadWrite<Word: Copy = u8>: Read<Word> + Write<Word> {
+    /// Writes and reads simultaneously. `write` is written to the slave on MOSI and
+    /// words received on MISO are stored in `read`.
+    ///
+    /// It is allowed for `read` and `write` to have different lengths, even zero length.
+    /// The transfer runs for `max(read.len(), write.len())` words. If `read` is shorter,
+    /// incoming words after `read` has been filled will be discarded. If `write` is shorter,
+    /// the value of words sent in MOSI after all `write` has been sent is implementation-defined,
+    /// typically `0x00`, `0xFF`, or configurable.
+    fn transfer(&mut self, read: &mut [Word], write: &[Word]) -> Result<(), Self::Error>;
```

Comment:
another small bikeshed... if we have `TransferInPlace` and `transfer_in_place` would it make sense to have `TransferSplit` and `transfer_split` as the reciprocal?

```suggestion
    fn transfer_split(&mut self, read: &mut [Word], write: &[Word]) -> Result<(), Self::Error>;
```

Reply:
The name was `transfer` before. IMO separate-buffers transfer is the "normal" one, in-place is the "weird" one, so I like the `transfer` name more.

Reply:
At first sight `transfer_split` sounds like the transfer itself is somehow split, which is rather confusing. I would also keep the `transfer` name.

Reply:
:+1: